src/plugins/dashboard/public/dashboard_container/embeddable/dashboard_container.tsx

```diff
@@ -0,0 +1,600 @@
+              <DashboardViewport onDataLoaded={this.onDataLoaded.bind(this)} container={this} />
+            </DashboardReduxWrapper>
+          </KibanaThemeProvider>
+        </ExitFullScreenButtonKibanaProvider>
+      </I18nProvider>,
+      dom
+    );
+  }
+
+  protected getInheritedInput(id: string): InheritedChildInput {
+    const {
+      query,
+      filters,
+      viewMode,
+      timeRange,
+      timeslice,
+      syncColors,
+      syncTooltips,
+      hidePanelTitles,
+      searchSessionId,
+      refreshInterval,
+      executionContext,
+    } = this.input as DashboardContainerByValueInput;
+
+    let combinedFilters = filters;
+    if (this.controlGroup) {
+      combinedFilters = combineDashboardFiltersWithControlGroupFilters(filters, this.controlGroup);
+    }
+    return {
+      refreshConfig: refreshInterval,
+      filters: combinedFilters,
+      hidePanelTitles,
+      searchSessionId,
+      executionContext,
+      syncTooltips,
+      syncColors,
+      timeRange,
+      timeslice,
+      viewMode,
+      query,
+      id,
+    };
+  }
+
+  // ------------------------------------------------------------------------------------------------------
+  // Cleanup
+  // ------------------------------------------------------------------------------------------------------
+  private stopDiffingDashboardState?: () => void;
+  private stopSyncingWithUnifiedSearch?: () => void;
+  private dataViewsChangeSubscription?: Subscription = undefined;
+  private stopSyncingDashboardSearchSessions: (() => void) | undefined;
+
+  public destroy() {
+    super.destroy();
+    this.onDestroyControlGroup?.();
+    this.subscriptions.unsubscribe();
+    this.stopDiffingDashboardState?.();
+    this.reduxEmbeddableTools?.cleanup();
+    this.stopSyncingWithUnifiedSearch?.();
+    this.stopSyncingDashboardSearchSessions?.();
+    this.dataViewsChangeSubscription?.unsubscribe();
+    if (this.domNode) ReactDOM.unmountComponentAtNode(this.domNode);
+  }
+
+  // ------------------------------------------------------------------------------------------------------
+  // Dashboard API
+  // ------------------------------------------------------------------------------------------------------
+
+  /**
+   * Sometimes when the ID changes, it's due to a clone operation, or a save as operation. In these cases,
+   * most of the state hasn't actually changed, so there isn't any reason to destroy this container and
+   * load up a fresh one. When an id change is in progress, the renderer can check this method, and if it returns
+   * true, the renderer can safely skip destroying and rebuilding the container.
+   */
+  public isExpectingIdChange() {
+    return this.expectingIdChange;
+  }
+  private expectingIdChange = false;
+  public expectIdChange() {
+    // this.expectingIdChange = true; TODO - re-enable this for saving speed-ups. It causes some functional test failures because the _g param is not carried over.
```

Comment:
Very true, this should be re-enabled. I figured it would be good to do this as a follow-up because it's really just a speed up / for efficiency purposes. Tracked in [this issue ](https://github.com/elastic/kibana/issues/147491) and I've linked to the issue in the code.